src/pi/SmartWebSocket.ts

```diff
@@ -0,0 +1,57 @@
+import WebSocket from './WebSocket';
+
+export interface Message {
+  [index: string]: any;
+}
+
+function awaitWebsocketReady(websocket: WebSocket) {
+  return new Promise((resolve, reject) => {
+    if (websocket.readyState === WebSocket.OPEN) {
+      resolve();
+      return;
+    }
+
+    websocket.addEventListener('open', () => {
+      resolve();
+    });
```

Comment:
This is by design for the moment, I'm intentionally not adding an error listener to the websocket because I don't know how we want to consume errors at the moment.

In its current state, the errors just propagate to the browser console which is pretty useful. If we reject here, we'll need to handle the rejection in the `SmartWebSocket`.

I think some kind of error handling is useful, but we should determine how this thing is going to be used, first.